databuilder/databuilder/extractor/dashboard/databricks_sql/databricks_sql_dashboard_extractor.py

```diff
@@ -0,0 +1,190 @@
+# Copyright Contributors to the Amundsen project.
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import (
+    Any, Iterator, Optional,
+)
+
+from pyhocon import ConfigFactory, ConfigTree
+
+from databuilder.extractor.base_extractor import Extractor
+from databuilder.extractor.dashboard.databricks_sql.databricks_sql_dashboard_utils import (
+    DatabricksSQLPaginatedRestApiQuery, generate_dashboard_description, get_text_widgets, get_visualization_widgets,
+    sort_widgets,
+)
+from databuilder.extractor.restapi.rest_api_extractor import REST_API_QUERY, RestAPIExtractor
+from databuilder.models.dashboard.dashboard_chart import DashboardChart
+from databuilder.models.dashboard.dashboard_last_modified import DashboardLastModifiedTimestamp
+from databuilder.models.dashboard.dashboard_metadata import DashboardMetadata
+from databuilder.models.dashboard.dashboard_owner import DashboardOwner
+from databuilder.models.dashboard.dashboard_query import DashboardQuery
+from databuilder.rest_api.base_rest_api_query import EmptyRestApiQuerySeed
+from databuilder.rest_api.rest_api_query import RestApiQuery
+from databuilder.transformer.base_transformer import ChainedTransformer
+from databuilder.transformer.timestamp_string_to_epoch import FIELD_NAME as TS_FIELD_NAME, TimestampStringToEpoch
+
+
+class DatabricksSQLDashboardExtractor(Extractor):
+    """
+    An extractor for retrieving dashboards, queries, and visualizations
+    from Databricks SQL (https://databricks.com/product/databricks-sql)
+    """
+
+    DATABRICKS_HOST_KEY = "databricks_host"
+    DATABRICKS_API_TOKEN_KEY = "databricks_api_token"
+
+    PRODUCT = "databricks-sql"
+    DASHBOARD_GROUP_ID = "databricks-sql"
+    DASHBOARD_GROUP_NAME = "Databricks SQL"
+
+    def init(self, conf: ConfigTree) -> None:
+        # Required configuration
+        self._databricks_host = conf.get_string(
+            DatabricksSQLDashboardExtractor.DATABRICKS_HOST_KEY
+        )
+        self._databricks_api_token = conf.get_string(
+            DatabricksSQLDashboardExtractor.DATABRICKS_API_TOKEN_KEY
+        )
+
+        # NOTE: The dashboards api is currently in preview. When it gets moved out of preview
+        # this will break and it will need to be changed
+        self._databricks_sql_dashboards_api_base = (
+            f"{self._databricks_host}/api/2.0/preview/sql/dashboards"
```

Comment:
actually does DBSQL support the same API as redash does?

Reply:
I'm not familiar enough with the redash API to say for certain but given how I mostly used the redash extractor for this it definitely looks like it. I still think it should be two separate extractors though cause developers probably want to be deliberate in which one they are using.

Reply:
make sense